addons/hr_holidays/__init__.py

```diff
@@ -5,3 +5,17 @@
 from . import models
 from . import report
 from . import wizard
+
+from odoo import api, SUPERUSER_ID
+
+def _hr_holiday_post_init(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    country_code = env.company.country_id.code
+    if country_code:
+        module_ids = False
+
+        # if the company is in France
+        if country_code == 'FR':
+            module_ids = env['ir.module.module'].search([('name', '=', 'l10n_fr_hr_work_entry_holidays'), ('state', '=', 'uninstalled')])
+        if module_ids:
+            module_ids.sudo().button_install()
```

Comment:
This will only take the default company, but I guess the module should be installed when there is *at least* one French company :thinking: 

Instead I'd make a search for companies with FR country code and then install the module :-)
```suggestion
    french_companies = env['res.company'].search([('country_id.code', '=', 'FR')])
    if french_companies:
        env['ir.module.module'].search([
            ('name', '=', 'l10n_fr_hr_work_entry_holidays'),
            ('state', '=', 'uninstalled')
        ]).sudo().button_install()
```